tools/capture_radar.rb

```diff
@@ -0,0 +1,49 @@
+Metric
+class Metric
+  def derived_cpu_total_cores_used
+    return if cpu_usage_rate_average.nil? || derived_vm_numvcpus.nil? || derived_vm_numvcpus == 0
+    (cpu_usage_rate_average * derived_vm_numvcpus) / 100.0
+  end
+end
+
+CustomAttribute
+class CustomAttribute
+  has_many   :metric_rollups, :as => :resource
+end
+
+labels = CustomAttribute.where(:name => "com.redhat.component")
+# Question - If a container is short lived, will we still be able to find it here?
```

Comment:
With targeted refresh using watches (landing in 4.6.1) the shortlived container will be refreshed, then if it's archived, the  CustomAttribute will stay here. Otherwise there is a small-big chance (based on the env size) the shortlived container will not be refreshed.

But, CustomAttribute will contain all archived Metrics, so this algorithm will take more and more time. Best would be to  scan attributes only of the Containers that are active in the <start, end> period we are scanning

Reply:
> CustomAttribute will contain all archived Metrics

What does that mean?  Metrics are not tied to CustomAttributes.

Reply:
@Fryguy we do

```labels.each_with_object(Hash.new { |h,k| h[k] = [] }) { |l, h| h[l] += l.resource.containers.to_a }```

When a container is archived, the label is archived with it. So CustomAttribute contains all labels of all active and archived containers and their metrics. In production we saw 10-20k containers being archived per month (while having 10-20k active), each with ~20 labels. So this table can bloat quickly.